### PR TITLE
[ROCm] Hipblas-lt integration: the Tensorflow part (2nd attempt)

### PR DESCRIPTION
**Pull request by pemeliya**
This is a remaining part of hipblas-lt integration: diffs are taken from the original pull request: https://github.com/ROCmSoftwarePlatform/tensorflow-upstream/pull/2000

The XLA part was already integrated in: https://github.com/openxla/xla/pull/3953/files#

@akuegel: can you please review the changes ?

### CODE REVIEW

File: tensorflow/core/kernels/matmul_op_test.cc [hunk 1]
```diff
@@ -221,6 +225,7 @@ class FusedMatMulOpTest : public OpsTestBase {
   // FusedMatMul.
   void VerifyMatMulWithBias(int m, int k, int n, bool transpose_a,
                             bool transpose_b) {
+    printf("=== VerifyMatMulWithBias ( %d, %d, %d, %d, %d ) ===\n", m, k, n, (int)transpose_a, (int)transpose_b);
```
Comment (akuegel) [inline tensorflow/core/kernels/matmul_op_test.cc:228]: Please change this to a VLOG.

File: tensorflow/core/kernels/matmul_op_test.cc [hunk 2]
```diff
@@ -281,21 +286,23 @@ TYPED_TEST_SUITE_P(FusedMatMulWithBiasOpTest);
 // -------------------------------------------------------------------------- //
 
 TYPED_TEST_P(FusedMatMulWithBiasOpTest, MatMul256x256x256) {
-  this->VerifyMatMulWithBias(256, 256, 256, false, false);
-  this->VerifyMatMulWithBias(256, 256, 256, true, false);
-  this->VerifyMatMulWithBias(256, 256, 256, false, true);
-  this->VerifyMatMulWithBias(256, 256, 256, true, true);
+  this->VerifyMatMulWithBias(256, 128, 64, false, false);
```
Comment (akuegel) [inline tensorflow/core/kernels/matmul_op_test.cc:289]: Why the change from 256 x 256 to 128 x 64?
Also, if this is necessary, then please also rename the test name accordingly.

File: tensorflow/core/kernels/matmul_op_test.cc [hunk 3]
```diff
@@ -307,13 +314,13 @@ TYPED_TEST_P(FusedMatMulWithBiasOpTest, MatMul1x256x1) {
 
 TYPED_TEST_P(FusedMatMulWithBiasOpTest, MatMul256x256x256WithActivation) {
   for (const string& activation : {"Relu", "Relu6", "Elu", "LeakyRelu"}) {
-    this->VerifyConv2DWithBiasAndActivation(256, 256, 256, false, false,
+    this->VerifyConv2DWithBiasAndActivation(256, 128, 64, false, false,
```
Comment (akuegel) [inline tensorflow/core/kernels/matmul_op_test.cc:317]: Same here, why the change of dimension sizes? And if necessary, then also update the test name.

File: tensorflow/core/util/autotune_maps/conv_parameters.cc [hunk 4]
```diff
@@ -94,7 +94,7 @@ MatmulParameters::MatmulParameters(
   proto_.set_c_dtype(c_dtype);
 
   proto_.set_trans_a(trans_a);
-  proto_.set_trans_b(trans_a);
+  proto_.set_trans_b(trans_b);
```
Comment (akuegel) [inline tensorflow/core/util/autotune_maps/conv_parameters.cc:97]: This looks like a bug fix that can land in a separate PR, together with a regression test?

File: tensorflow/core/kernels/matmul_op_test.cc (hunk 2 above)
Comment (pemeliya) [reply tensorflow/core/kernels/matmul_op_test.cc:289]: yes, I need to check it out with Eugene when he is available, it might be that this is only necessary for the ROCm platform

File: tensorflow/core/util/autotune_maps/conv_parameters.cc (hunk 4 above)
Comment (pemeliya) [reply tensorflow/core/util/autotune_maps/conv_parameters.cc:97]: makes sense, I can remove this change and we will add it as a separate PR

File: tensorflow/core/kernels/matmul_op_test.cc (hunk 2 above)
Comment (pemeliya) [reply tensorflow/core/kernels/matmul_op_test.cc:289]: According to Eugene, this was done in order to make sure that the matrix dimensions are handled in a correct order. Otherwise, if one accidentally permutes 256x256x256, this won't be caught by the test, I will rename the test(s) accordingly

**Review by akuegel — APPROVED** [review]

**Review by akuegel — APPROVED** [review]

**Comment by akuegel** [comment]
It seems there is one test failure for this test:

https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/ops/batch_ops_test.py

@pemeliya Can you please check whether you can reproduce this?

**Comment by akuegel** [comment]
Thanks for checking. It seems this was a flaky test, now it seems to pass.

File: tensorflow/core/kernels/BUILD [hunk 5]
```diff
@@ -41,6 +41,8 @@ load(
 load(
     "@local_config_rocm//rocm:build_defs.bzl",
     "if_rocm",
+    "rocm_hipblaslt",
```
Comment (cantonios) [inline tensorflow/core/kernels/BUILD:44]: This is apparently unused, and our BUILD file checker is asking for it to be removed.

**Review by cantonios — CHANGES_REQUESTED** [review]

File: tensorflow/core/kernels/BUILD (hunk 5 above)
Comment (pemeliya) [reply tensorflow/core/kernels/BUILD]: sorry, overlooked this comment. Now I rebased the branch and fixed this

**Review by cantonios — APPROVED** [review]
